### PR TITLE
[4.2] Add autocomplete=one-time-code attribute to mfa input field

### PR DESCRIPTION
**Pull request by SniperSister**
### Summary of Changes
Added the autocomplete attribute to the mfa input field on the captive page. By default, "one-time-code" is used as the attribute value, but plugins can override it if necessary - see the "fixed" plugin, which set's the attribute to "off" (even though that setting will be ignored by many modern browsers, so consider this as a polite suggestion for the browser...).

As discussed with Nic, I'm doing this as a public PR as MFA codes are one-time-codes and therefore even a previously stored MFA token isn't a security issue.

### Testing Instructions
Setup a 4.2 site, configure MFA for an example user, verify the existence of the attribute on the MFA input field on the captive pager after login.

### Actual result BEFORE applying this Pull Request
No attribute.

### Expected result AFTER applying this Pull Request
Attribute.

### CODE REVIEW

**Comment by SniperSister** [comment]
Please do not test the PR yet, I have to take care of an edge case first...

**Comment by brianteeman** [comment]
_as this is in tmpl view files that may be overridden in a template we need to be very careful with the change to learn the lessons from the past and to ensure that this change does not break any existing overrides. In addition as the change is a security improvement I strongly recommend that the change is well documented on release to ensure that people with an override are aware of the importance of updating their override._

**Comment by SniperSister** [comment]
> as this is in tmpl view files that may be overridden in a template we need to be very careful with the change to learn the lessons from the past and to ensure that this change does not break any existing overrides. 

It does not break existing overrides. It's an additional attribute which simply won't be rendered in older overrides.

> In addition as the change is a security improvement I strongly recommend that the change is well documented on release to ensure that people with an override are aware of the importance of updating their override.

Will do!

**Comment by brianteeman** [comment]
> > as this is in tmpl view files that may be overridden in a template we need to be very careful with the change to learn the lessons from the past and to ensure that this change does not break any existing overrides.
> 
> It does not break existing overrides. It's an additional attribute which simply won't be rendered in older overrides.


Said every joomla developer ever

File: administrator/components/com_users/tmpl/captive/default.php [hunk 1]
```diff
@@ -78,12 +78,13 @@ class="form-horizontal"
...
-                        'placeholder' => $this->renderOptions['placeholder'] ?? null,
-                        'id'          => 'users-mfa-code',
-                        'class'       => 'form-control'
+                        'type'         => $this->renderOptions['input_type'],
+                        'name'         => 'code',
+                        'value'        => '',
+                        'placeholder'  => $this->renderOptions['placeholder'] ?? null,
+                        'id'           => 'users-mfa-code',
+                        'class'        => 'form-control',
+                        'autocomplete' => (!empty($this->renderOptions['autocomplete'])) ?  $this->renderOptions['autocomplete'] : 'one-time-code'
```
Comment (Quy) [inline administrator/components/com_users/tmpl/captive/default.php]:
```suggestion
                        'autocomplete' => !empty($this->renderOptions['autocomplete']) ? $this->renderOptions['autocomplete'] : 'one-time-code'
```

File: components/com_users/tmpl/captive/default.php [hunk 2]
```diff
@@ -81,12 +81,13 @@ class="form-horizontal"
...
-                            'placeholder' => $this->renderOptions['placeholder'] ?? null,
-                            'id'          => 'users-mfa-code',
-                            'class'       => 'form-control'
+                            'type'         => $this->renderOptions['input_type'],
+                            'name'         => 'code',
+                            'value'        => '',
+                            'placeholder'  => $this->renderOptions['placeholder'] ?? null,
+                            'id'           => 'users-mfa-code',
+                            'class'        => 'form-control',
+                            'autocomplete' => (!empty($this->renderOptions['autocomplete'])) ?  $this->renderOptions['autocomplete'] : 'one-time-code'
```
Comment (Quy) [inline components/com_users/tmpl/captive/default.php]:
```suggestion
                            'autocomplete' => !empty($this->renderOptions['autocomplete']) ? $this->renderOptions['autocomplete'] : 'one-time-code'
```

File: administrator/components/com_users/tmpl/captive/default.php [hunk 3]
```diff
@@ -78,12 +78,13 @@ class="form-horizontal"
...
-                        'placeholder' => $this->renderOptions['placeholder'] ?? null,
-                        'id'          => 'users-mfa-code',
-                        'class'       => 'form-control'
+                        'type'         => $this->renderOptions['input_type'],
+                        'name'         => 'code',
+                        'value'        => '',
+                        'placeholder'  => $this->renderOptions['placeholder'] ?? null,
+                        'id'           => 'users-mfa-code',
+                        'class'        => 'form-control',
+                        'autocomplete' => !empty($this->renderOptions['autocomplete']) ? $this->renderOptions['autocomplete'] : 'one-time-code'
```
Comment (HLeithner) [inline administrator/components/com_users/tmpl/captive/default.php]:
```suggestion
                        'autocomplete' => $this->renderOptions['autocomplete'] ?? 'one-time-code'
```

File: components/com_users/tmpl/captive/default.php [hunk 4]
```diff
@@ -81,12 +81,13 @@ class="form-horizontal"
...
-                            'placeholder' => $this->renderOptions['placeholder'] ?? null,
-                            'id'          => 'users-mfa-code',
-                            'class'       => 'form-control'
+                            'type'         => $this->renderOptions['input_type'],
+                            'name'         => 'code',
+                            'value'        => '',
+                            'placeholder'  => $this->renderOptions['placeholder'] ?? null,
+                            'id'           => 'users-mfa-code',
+                            'class'        => 'form-control',
+                            'autocomplete' => !empty($this->renderOptions['autocomplete']) ? $this->renderOptions['autocomplete'] : 'one-time-code'
```
Comment (HLeithner) [inline components/com_users/tmpl/captive/default.php]:
```suggestion
                            'autocomplete' => $this->renderOptions['autocomplete'] ?? 'one-time-code'
```

**Comment by HLeithner** [comment]
rebased for 4.3

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.0-dev. No new features will be merged into Joomla! 4.3 series. Joomla! 4.4 series is a bridge release to make migration from Joomla! 4 to 5 as smooth as possible.

**Comment by HLeithner** [comment]
I think it's save for joomla 5.0, thanks
